src/refda/exec.c

```diff
@@ -822,3 +805,59 @@ int refda_exec_sbr_disable(refda_agent_t *agent, refda_amm_sbr_desc_t *sbr)
     atomic_fetch_sub(&agent->instr.num_sbrs, 1);
     return 0;
 }
+
+/** Execute an action that has already been verified
+ * Based on code from refda_exec_exp_execset
+ */
+static int refda_exec_action(refda_agent_t *agent, refda_exec_seq_t *seq, const cace_ari_t *action)
+{
+    refda_runctx_ptr_t ctxptr;
+    refda_runctx_ptr_init_new(ctxptr);
+
+    refda_runctx_t *runctx = refda_runctx_ptr_ref(ctxptr);
+
+    if (refda_runctx_from(runctx, agent, NULL))
+    {
+        return 2;
+    }
+
+    refda_runctx_ptr_set(seq->runctx, ctxptr);
+    int res = refda_exec_exp_mac(runctx, seq, action);
+
+    refda_runctx_ptr_clear(ctxptr); // Clean up extra reference created by ptr_ref
+    return res;
+}
+
+int refda_exec_queue(refda_agent_t *agent, const cace_ari_t *ari)
```

Comment:
A reason why this implementation was delayed was in trying to integrate the sub-execution (from a target in a CTRL actual parameter) into to main execution pipeline. Is this method here compatible with the pipeline? Meaning can this execution be statused by EDD and possibly interrupted by other CTRL?

Reply:
@BrianSipos Please let me know your thoughts on the above, and whether it would be better to take the discussion offline.